src/sirocco/parsing/_yaml_data_models.py

```diff
@@ -70,10 +70,8 @@ def convert_datetime(cls, value) -> datetime:
         return datetime.fromisoformat(value)
 
 
-# TODO: Change class name, does not fit anymore wit hthe addition of `when` and `parameters`
-#       find something more related to graph specification in general like _GraphTargetBaseModel
-class _LagDateBaseModel(BaseModel):
-    """Base class for all classes containg a list of dates or time lags."""
+class TargetNodesBaseModel(_NamedBaseModel):
+    """class for targetting other task or data nodes in the graph"""
```

Comment:
Can you explain this a bit more in the docstring? Reading it, it is not so clear for me

Reply:
Maybe something like this. But I am also not really happy with this
```suggestion
    """Class for specifying time information relative to the cycle date, and
    information related to the parametrization. """
```
I hope when we move to a release things become clearer